akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingSpec.scala

```diff
@@ -766,16 +764,28 @@ abstract class ClusterShardingSpec(multiNodeConfig: ClusterShardingSpecConfig)
       }
       enterBarrier("persistent-started")
 
+      // watch-out, these two var are only init on 3rd node
+      var shard: ActorSelection = null
+      var region: ActorSelection = null
       runOn(third) {
         //Create an increment counter 1
         persistentEntitiesRegion ! EntityEnvelope(1, Increment)
         persistentEntitiesRegion ! Get(1)
         expectMsg(1)
 
-        //Shut down the shard and confirm it's dead
-        val shard = system.actorSelection(lastSender.path.parent)
-        val region = system.actorSelection(lastSender.path.parent.parent)
+        shard = system.actorSelection(lastSender.path.parent)
+        region = system.actorSelection(lastSender.path.parent.parent)
+      }
+      enterBarrier("counter-incremented")
+
+      // clean up shard cache everywhere
+      runOn(third, fourth, fifth) {
+        persistentEntitiesRegion ! BeginHandOff("1")
+        expectMsg(10 seconds, "ShardStopped not received", BeginHandOffAck("1"))
+      }
+      enterBarrier("everybody-hand-off-ack")
```

Comment:
@patriknw, I took another approach here. Instead of mimicking how the RebalanceWork works, I'm just sending `BeginHandOff` to all regions so cache is cleared. Then I continue as before and send the `HandOff` to the region holding the shard. 

I run it locally and worked. Travis and Jenkins won't these test, as far I can tell.